ext/opentelemetry-ext-requests/src/opentelemetry/ext/requests/__init__.py

```diff
@@ -89,9 +88,9 @@ def instrumented_request(self, method, url, *args, **kwargs):
 
         exception = None
 
-        with tracer.start_as_current_span(
-            span_name, kind=SpanKind.CLIENT
-        ) as span:
+        with get_tracer(
```

Comment:
thoughts on making this a variable that's lazily created, rather than creating the trace every time?

I worry about the overhead of doing so. Very low right now but not sure about the future.

Reply:
@ocelotl can you take a quick look at this comment? then we can merge.

Reply:
Hm, getting the tracer every time is intentional, this is what makes these changes work, as mentioned above.

Reply:
I think technically it's not getting the tracer every time: it's just deferring tracer creation until after configuration.

other instrumentations (see starlette for example) work well with auto-instrumentation as long as the configuration is performed *before* an app is instantiated.

But I don't think I'll argue the point further. We can refactor this a bit more when there's a strong reason to memoize tracers.